engine-tests/src/main/java/org/terasology/rendering/GLSLShaderTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.rendering;
+
+import java.awt.GraphicsEnvironment;
+import java.util.Collections;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.terasology.DisplayEnvironment;
+import org.terasology.asset.AssetType;
+import org.terasology.asset.AssetUri;
+import org.terasology.rendering.assets.shader.ShaderData;
+import org.terasology.rendering.opengl.GLSLShader;
+
+/**
+ * Tests the {@link GLSLShader} class.
+ */
+public class GLSLShaderTest {
+
+    @BeforeClass
+    public static void setupClass() {
+        // don't run any of the tests in this class if environment has no graphics support
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        new DisplayEnvironment();
+    }
+
+    @Test
+    public void testOk() {
```

Comment:
I have very little experience with unit testing and perhaps these are typical test names. If the aren't however, I'd rename testOk() and testBroken() to testConstructorWithValidShader() and testConstructorWithFaultyShader().

Also, is it worth testing what happens if a non AssetType.SHADER uri is provided?